fedimint-core/src/lib.rs

```diff
@@ -146,25 +156,43 @@ pub fn sats(amount: u64) -> Amount {
     Amount::from_sats(amount)
 }
 
+pub mod amount {
+    pub mod serde {
+        pub mod as_sat {
+            //! Serialize and deserialize [`Amount`](crate::Amount) as integers
+            //! denominated in satoshi. Use with
+            //! `#[serde(with = "amount::serde::as_sat")]`.
+
+            use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+            pub fn serialize<S: Serializer>(a: &crate::Amount, s: S) -> Result<S::Ok, S::Error> {
+                u64::serialize(&a.sats(), s)
+            }
+
+            pub fn deserialize<'d, D: Deserializer<'d>>(d: D) -> Result<crate::Amount, D::Error> {
+                Ok(crate::Amount::from_sats(u64::deserialize(d)?))
+            }
+        }
+    }
+}
+
 /// Amount of bitcoin to send, or "all" to send all available funds
 #[derive(Debug, Eq, PartialEq, Copy, Hash, Clone, Serialize, Deserialize)]
 #[serde(rename_all = "snake_case")]
 pub enum BitcoinAmountOrAll {
```

Comment:
If we want to use `AmountOrAll` for amounts with different precisions we should make it generic over the amount type. Converting back and forth between different precisions can lead to all kinds of problems:

* APIs dealing with sat precision now silently throw away the msat part of the amount instead of returning an error
* APIs dealing with msat precision might accidentally lose it due to how serialization is implemented in this PR